dongle/mt76.cpp

```diff
@@ -1408,6 +1408,31 @@ bool Mt76::initChannels()
     return true;
 }
 
+bool Mt76::openFirmware()
+{
+    const std::string filename(DATADIR "/FW_ACC_00U.bin");
+
+    if (firmware.size() != 0)
+    {
+        Log::debug("Firmware binary already loaded.");
+        return true;
+    }
+
+    std::ifstream blob(filename, std::ios::binary);
+    if (!blob)
+    {
+        Log::error("Failed to open firmware binary %s\n\n"
+		"You can get the firmware using: xow-get-firmware.sh\n",
+		filename.c_str());
+        return false;
+    }
+    std::istream_iterator<uint8_t> start(blob), end;
+    noskipws(blob);
+
+    firmware = Bytes(std::vector<uint8_t>(start, end));
```

Comment:
practically speaking Bytes has its own std::vector, so perhaps just writing into a plain buffer and passing it over is more efficent, since we don't need to keep the data around

Reply:
Not sure I get that. Can you add a code suggestion?

Reply:
```c
    uint8_t* firmware_buf = new uint8_t[size];
    if (!blow.read((char*)firmware_buf, size))
    {
        Log::error("Failed to load firmware into memory");
        delete[] firmware_buf;

        return false;
    }

    const Bytes firmware(
        firmware_buf,
        firmware_buf+size
    );
```
[code](https://github.com/Jan200101/xow/blob/16ac5b3d6c6f8c6494324946b9866d5858674536/dongle/mt76.cpp#L1464-L1476)
This was from my attempt at this before I saw this PR existed and uses more primitive types to get the job done but honestly this is more of a personal gripe than anything.

it also adds another check for actually reading the file, since there is the possibility that the file we have opened disappears because e.g. its on a portable medium